### PR TITLE
call vim.fn.complete when event loop is processing GUI event

### PR DESCRIPTION
**Pull request by RishabhRD**
Hello @hrsh7th, Opening a PR so we can discuss in terms of code. #61 shows
weird backspace behavior.

I feel `vim.fn.complete` should be called when the GUI event is taking place in
nvim-event-loop. This PR currently adds `vim.schedule` wrapper around it. At least
this solves it in my machine. What is the effect on your environment?

If not solves in your environment I can change work over code more as I am
interested in this cool project. :smile:

### CODE REVIEW

**Comment by hrsh7th** [comment]
Thank you for your contribution!

I thought almost logic is safe to neovim's fast_event, in nvim-compe...

File: lua/compe/completion.lua [hunk 1]
```diff
@@ -231,7 +231,9 @@ end
 Completion._show = function(start_offset, items)
```
Comment (hrsh7th) [inline lua/compe/completion.lua:231]: I think we should wrap all `Completion._show` codes by `vim.schedule`.

**Comment by hrsh7th** [comment]
This PR improve behavior!

File: lua/compe/completion.lua (hunk 1 above)
Comment (RishabhRD) [reply lua/compe/completion.lua:231]: Yeah! This would make the behavior more predictable in terms of execution.

**Comment by RishabhRD** [comment]
@hrsh7th I just pushed the code to wrap the function in vim.schedule, I guess
this is what you were talking about in the code review. Or I got it wrong? :sweat_smile:

**Comment by hrsh7th** [comment]
No. You are right. LGTM!

**Pull request merged**